src/rgw/rgw_common.h

```diff
@@ -1690,6 +1690,7 @@ struct req_state : DoutPrefixProvider {
   std::vector<rgw::IAM::Policy> session_policies;
 
   jspan trace;
+  bool trace_enabled;
```

Comment:
this variable is uninitialized

couldn't this just be a local variable in `process_request()`?

Reply:
right
thanks

Reply:
you can probably remove it from here

Reply:
no sorry, I was wrong. we should keep it in `req_state` and not as local var. because it's a member of `req_state` we can get and change it inside the lua script execution.

Reply:
then please initialize it here: `bool trace_enabled = false;`

Reply:
done